libindy-crypto/src/bls/mod.rs

```diff
@@ -335,7 +358,34 @@ impl Bls {
     /// assert!(valid);
     /// ```
     pub fn verify(signature: &Signature, message: &[u8], ver_key: &VerKey, gen: &Generator) -> Result<bool, IndyCryptoError> {
-        let h = Bls::_hash(message)?;
+        let h = Bls::_hash_sha2(message)?;
+        Ok(Pair::pair(&signature.point, &gen.point)?.eq(&Pair::pair(&h, &ver_key.point)?))
```

Comment:
Similar argument as above, abstract `Pair::pair(&signature.point, &gen.point)?.eq(&Pair::pair(&<message hash>, &ver_key.point)` in a different function or explicitly pass hash function as an arg or better make `verify_pop` part of `Verkey`